src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1931,11 +1947,24 @@ private Conversion GenerateConversionForConditionalOperator(BoundExpression sour
         private static Conversion GenerateConversion(Conversions conversions, BoundExpression sourceExpression, TypeSymbol sourceType, TypeSymbol destinationType)
         {
             HashSet<DiagnosticInfo> useSiteDiagnostics = null;
-            return useExpressionForConversion(sourceExpression) ?
+            return UseExpressionForConversion(sourceExpression) ?
                 conversions.ClassifyImplicitConversionFromExpression(sourceExpression, destinationType, ref useSiteDiagnostics) :
                 conversions.ClassifyImplicitConversionFromType(sourceType, destinationType, ref useSiteDiagnostics);
+        }
 
-            bool useExpressionForConversion(BoundExpression value) => value.Type is null || value.ConstantValue != null;
+        private static bool UseExpressionForConversion(BoundExpression value)
+        {
+            if (value.Type is null || value.Type.IsDynamic() || value.ConstantValue != null)
+            {
+                return true;
+            }
+            switch (value.Kind)
+            {
+                case BoundKind.InterpolatedString:
```

Comment:
Is there an test covering the interpolated string path here?

Reply:
`StaticNullChecking.StringInterpolation_01`

---
In reply to: [183829916](https://github.com/dotnet/roslyn/pull/26356#discussion_r183829916) [](ancestors = 183829916)